sw/device/silicon_owner/tock/tests/uart/BUILD

```
@@ -0,0 +1,50 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+load("@rules_rust//rust:defs.bzl", "rust_binary")
+load("//rules/opentitan:defs.bzl", "OPENTITAN_CPU", "cw310_params", "opentitan_test")
+load("//rules:tock.bzl", "tock_elf2tab", "tock_image")
+
+package(default_visibility = ["//visibility:public"])
+
+rust_binary(
+    name = "uart",
+    srcs = [
+        "src/main.rs",
+    ],
+    # We specifically restrict our build target to the OpenTitan
+    # CPU because libtock does not support an x86_64 target.
+    target_compatible_with = [OPENTITAN_CPU],
+    deps = [
+        "//sw/device/silicon_owner/tock/apps:single_app_layout",
+        "@libtock",
+    ],
+)
+
+tock_elf2tab(
+    name = "tab",
+    src = ":uart",
+    arch = "rv32imc",
+)
+
+tock_image(
+    name = "image",
+    app_flash_start = 0x20040000,
+    apps = [":tab"],
+    kernel = "//sw/device/silicon_owner/tock/kernel",
+)
+
+opentitan_test(
+    name = "uart_test",
+    exec_env = {
+        "//hw/top_earlgrey:fpga_hyper310_test_rom": None,
+    },
+    hyper310 = cw310_params(
```

Consider using one of the existing environments:

```suggestion
    exec_env = {
        "//hw/top_earlgrey:fpga_cw310_sival_rom_ext": None,
        "//hw/top_earlgrey:fpga_hyper310_rom_ext": "cw310",
    },
    cw310 = cw310_params(
```